pkg/apis/hostpathprovisioner/v1beta1/hostpathprovisioner_types.go

```diff
@@ -49,6 +53,26 @@ type HostPathProvisionerStatus struct {
 	ObservedVersion string `json:"observedVersion,omitempty" optional:"true"`
 }
 
+// VolumeSource defines a volume with a kind and a source
+// +k8s:openapi-gen=true
+type VolumeSource struct {
+	// Kind specifies an identifier that is used in the storage class arguments to identify the source to use.
+	Kind string `json:"kind" valid:"required"`
+	// SourceStorageClass specifies which storage class to use to create PersistVolumeClaims against and the template of the claim
+	SourceStorageClass *SourceStorageClass `json:"sourceStorageClass,omitempty" optional:"true"`
+	// PVC specifies the persistent volume claim to use. This is useful when you want one PVC to be shared among all nodes
```

Comment:
The shared storage use case is not part of this PR.  Might be better to avoid mentioning it here.  To me, PVC is useful only for single node clusters when the administrator wants more control over the provisioning of storage for HPP.  Come to think of it, the same could be achieved with SourceStorageClass so maybe we should omit PVC for now.

Also the name SourceStorageClass is not consistent with the names PVC and Path (which do not have a Source prefix).

Reply:
Right, I am not entirely sure I like the 'source' part of any of it, but for consistency remove it from the storageClass part completely. Thinking about it more, I am not liking volumeSource either, but not sure what else to call it.